### ISSUE TITLE
Adding dualstack option for webinterface - help needed

### ISSUE THREAD

**Issue by treysis** [pyload/pyload#3543]
### Your question

I am trying to add a dualstack option to circumvent the difficulties of parsing `::` as host address for the web interface. I have therefor made some modifications to some files:

**pyload.conf**
```
webinterface - "Webinterface":
    ...
    bool dualstack : "Dualstack" = True
    ...
```
**module/web/ServerThread.py**
```
        ...
        self.host = pycore.config['webinterface']['host']
        dualstack = pycore.config['webinterface']['dualstack']
        if self.host == "0.0.0.0" and dualstack:
            self.host = "::"
        self.port = pycore.config['webinterface']['port']
        ...
```

However, adding `pycore.config['webinterface']['dualstack']` results in the following error trace:
```
03.03.2020 00:01:19 ERROR     <<< UNCAUGHT EXCEPTION >>>
Traceback (most recent call last):
  File ".\pyLoadCore.py", line 687, in <module>
    main()
  File ".\pyLoadCore.py", line 678, in main
    pyload_core.start()
  File ".\pyLoadCore.py", line 422, in start
    self.init_webserver()
  File ".\pyLoadCore.py", line 500, in init_webserver
    self.webserver = WebServer(self)
  File "D:\pyload-0420\pyload\dist\module\web\ServerThread.py", line 25, in __init__
    dualstack = pycore.config['webinterface']['dualstack']
  File "D:\pyload-0420\pyload\dist\module\ConfigParser.py", line 376, in __getitem__
    return self.parser.get(self.section, item)
  File "D:\pyload-0420\pyload\dist\module\ConfigParser.py", line 283, in get
    val = self.config[section][option]["value"]
KeyError: 'dualstack'
```

Why doesn't it simply parse the additional line? Where is defined which config lines can be parsed? I fail to see this. I thought it's a simple solution...but now it seems it might be even easier to just patch `ConfigParser.py` to accept `::` as value for an option.

### Additional references

It seems the error is somehow related to Python's "threading.py" liberary. At least there I was able to find the "KeyError:" string. Still no idea what triggers this error exactly.
I am running Python 2.7.17.

**Comment by treysis** [comment]
Found and fixed it. The problem was that I forgot to also define it in "default.conf". However, it seems the default version of CherryPyWSGIServer (CherryPy/3.1.2) doesn't run dualstacked if host is set to `::` (even though documentation says otherwise). I updated CherryPy to 6.2.1, but I haven't checked if SSL support is working. Because with 8.9.1 SSL support seems broken.

**Comment by GammaC0de** [comment]
Please inform which version of CherryPy is best, I mightadd it to `stable`.

**Comment by treysis** [comment]
So, not sure from which version on it works correctly, but I decided to stick with the most recent CherryPy/8.9.1 (link is provided at the end). Most recent being 8.9.1 because the CherryPy project moved the webserver to a new project called cheroot and while this didn't change much, it changed the class name and I was bored to refactor everything.
Unfortunately, after 3.1.2 the implementation for using HTTPS/SSL changed. So if you want to keep HTTPS support, some additional changes are needed. My code looks now as follows (I always fail to manage git, so no PR, unfortunately 😐):
**ServerThread.py:** (from line 24)
```
self.host = pycore.config['webinterface']['host']
if self.host == "0.0.0.0" and pycore.config['webinterface']['dualstack']:
    self.host = "::"
self.port = pycore.config['webinterface']['port']
```
**webinterface.py:** (from line 145)
```
def run_threaded(host="0.0.0.0", port="8000", theads=3, cert="", key=""):
    from module.lib.wsgiserver import CherryPyWSGIServer
    from module.lib.wsgiserver.ssl_builtin import BuiltinSSLAdapter

    if cert and key:
        CherryPyWSGIServer.ssl_adapter = BuiltinSSLAdapter(cert, key)

    CherryPyWSGIServer.numthreads = theads

    from utils import CherryPyWSGI

    run(app=web, host=host, port=port, server=CherryPyWSGI, quiet=True)
```
**web/utils.py:** (in line 146)
`        from module.lib.wsgiserver import CherryPyWSGIServer`
**wsgiserver/ssl_builtin.py:** (in line 24)
`import wsgiserver`

Link to CherryPy/8.9.1 (chose CherryPy-8.9.1.tar.gz and just use folder wsgiserver):
https://pypi.org/project/CherryPy/8.9.1/#files

**Comment by treysis** [comment]
Parsing for "::" as IP address in the config file is very complicated. I therefor created a pull-request with my proposed solution from above, additionally I updated CherryPy to cheroot/5.11.0:
https://github.com/pyload/pyload/pull/3619

**Comment by GammaC0de** [comment]
Solved with #3619
